### DISCUSSION TITLE
Webassembly Multithreading + JS async interop in .net9 -- where are we?

### DISCUSSION THREAD

**Discussion by jtorjo** [dotnet/runtime#108915] [General]
Long story short, the more I read, the more confused I am -- what is supposed to work on .net9?
Also, some parts work in Blazor, some don't.

Say I have a blazor app.
@pavelsavara  On the main thread, I'm parsing some huge json, more or less like this:

`await Task.Run(() => long_consuming_task());`

will this work?

**Comment by pavelsavara** [comment]
Blazor didn't finish adopting threads in Net9 and so runtime doesn't consider it stable either.

**Comment by jtorjo** [comment]
Thanks @pavelsavara . So basically, when .net9 is declared stable, we still don't have stable multi-threading.

Are we 6 months away from stable webassembly multi-threading , or 1 year+ away?
Will we have it in some preview of .net10 or even later?
